modules/apps/27-interchain-accounts/host/keeper/keeper.go

```diff
@@ -102,6 +103,22 @@ func (k Keeper) GetActiveChannelID(ctx sdk.Context, connectionID, portID string)
 	return string(store.Get(key)), true
 }
 
+// GetOpenActiveChannel retrieves the active channelID from the store, keyed by the provided connectionID and portID & checks if the channel in question is in state OPEN
+func (k Keeper) GetOpenActiveChannel(ctx sdk.Context, connectionID, portID string) (string, bool) {
```

Comment:
Should we just change the `GetActiveChannelID` function to use this implementation (add the state check)? And use that throughout.

Afaik, the only place `GetActiveChannelID` is used is controller/relay.go in `SendTx`.
It seems most of the active channel checks are now using this approach with the additional state check, which kind of makes `GetActiveChanelID` redundant. 

Thoughts?

Reply:
I like the separation of concerns. Usually `Get` function retrieve values from the store. I'd find it odd if `GetActiveChannelID` also did a state check